forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-rcs/forgerock-openbanking-uk-aspsp-rs-rcs-server/src/test/java/com/forgerock/openbanking/aspsp/rs/rcs/api/rcs/decisions/RCSConsentDecisionApiControllerTest.java

```diff
@@ -0,0 +1,181 @@
+import com.forgerock.openbanking.aspsp.rs.rcs.services.RCSErrorService;
+import com.forgerock.openbanking.aspsp.rs.rcs.services.RcsService;
+import com.forgerock.openbanking.common.conf.RcsConfiguration;
+import com.forgerock.openbanking.common.constants.RCSConstants;
+import com.forgerock.openbanking.common.model.rcs.RedirectionAction;
+import com.forgerock.openbanking.common.model.rcs.consentdecision.ConsentDecision;
+import com.forgerock.openbanking.common.services.JwtOverridingService;
+import com.forgerock.openbanking.common.services.store.tpp.TppStoreService;
+import com.forgerock.openbanking.exceptions.OBErrorException;
+import com.forgerock.openbanking.jwt.services.CryptoApiClient;
+import com.forgerock.openbanking.model.Tpp;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.forgerock.openbanking.aspsp.rs.rcs.api.rcs.JwtTestHelper.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class RCSConsentDecisionApiControllerTest {
+
+    @Mock
+    private CryptoApiClient cryptoApiClient;
+    @Mock
+    private RcsService rcsService;
+    @Mock
+    private RcsConfiguration rcsConfiguration;
+    @Mock
+    private RCSErrorService rcsErrorService;
+    @Mock
+    private UserProfileService userProfileService;
+    @Mock
+    private IntentTypeService intentTypeService;
+    @Mock
+    private TppStoreService tppStoreService;
+    @Mock
+    private JwtOverridingService jwtOverridingService;
+
+    private final AMOpenBankingConfiguration amOpenBankingConfiguration = new AMOpenBankingConfiguration();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    // Class under test
+    private RCSConsentDecisionApiController consentDecisionApiController;
+
+
+    @Before
+    public void setUp() throws Exception {
+        this.consentDecisionApiController = new RCSConsentDecisionApiController(cryptoApiClient,rcsService,
+                rcsConfiguration,amOpenBankingConfiguration, rcsErrorService, objectMapper, userProfileService,
+                intentTypeService, tppStoreService, jwtOverridingService);
+
+        SinglePaymentConsentDecisionDelegate singlePaymentConsentDecisionDelegate =
+                mock(SinglePaymentConsentDecisionDelegate.class);
+        when(intentTypeService.getConsentDecision(anyString())).thenReturn(singlePaymentConsentDecisionDelegate);
```

Comment:
Minor - you're still using `when` here but `given` elsewhere.

Reply:
Fixed.